app/digital/ts/models/ioobjects/gates/XORGate.ts

```diff
@@ -26,6 +30,40 @@ export class XORGate extends Gate {
         return "or.svg";
     }
 
+    public getMinPos(): Vector {
+        const min = super.getMinPos();
+
+        var BOX_WIDTH = DEFAULT_BORDER_WIDTH;
+        if (this.numInputs() >3 && this.numInputs() < 7)
+            BOX_WIDTH += GATE_OR_CULLBOX_OFFSET;
+        if (this.numInputs() == 8)
+            BOX_WIDTH += GATE_OR_CULLBOX_OFFSET*2;
+
+        // Find minimum pos from corners of transform
+        const corners = this.transform.getCorners().map(
+            v => v.sub(DEFAULT_BORDER_WIDTH,BOX_WIDTH)
+        );
+
+        return Vector.min(min, ...corners);
+    }
+
+    public getMaxPos(): Vector {
+        const max = super.getMaxPos();
+
+        var BOX_WIDTH = DEFAULT_BORDER_WIDTH;
+        if (this.numInputs() >3 && this.numInputs() < 7)
+            BOX_WIDTH += GATE_OR_CULLBOX_OFFSET;
+        if (this.numInputs() == 8)
+            BOX_WIDTH += GATE_OR_CULLBOX_OFFSET*2;
```

Comment:
Given how many times you have to write this same block of code - we should put it in a utility function somewhere. 

Maybe in ORGate.ts we could make an `export function GetQuadraticOffset(numInputs: number)` under the ORGate class.